File: src/conf/plugins/coding/nvim-lint.ts
import { Plugin } from "@core/model";

export const plugin = new Plugin({
  shortUrl: "mfussenegger/nvim-lint",
  lazy: {
    event: ["BufReadPost", "BufNewFile"],
    config: () => {
      (luaRequire("lint").linters_by_ft as any) = {
        cpp: ["cpplint", "cppcheck"],
      };

      vim.api.nvim_create_autocmd(["BufWritePost"], {
        callback: () => {
          luaRequire("lint").try_lint();
        },
      });
    },
  },
});

File: lua/dora/packages/coding/plugins/nvim-lint.lua
---@type dora.core.plugin.PluginOption
return {
  "mfussenegger/nvim-lint",
  event = { "BufReadPost", "BufNewFile" },
  opts = {
    linters_by_ft = {},
  },
  config = function(_, opts)
    require("lint").linters_by_ft = opts.linters_by_ft

    vim.api.nvim_create_autocmd("BufWritePost", {
      function()
        require("lint").try_lint()
      end,
    })
  end,
}
